spampd.pl

```diff
@@ -898,6 +927,56 @@ sub setup_logging {
 
 ##################   SERVER METHODS   ######################
 
+sub audit {
+  my ($self, $msglines) = @_;
+  my $prop = $self->{spampd};
+  my $status;
+  # Audit the message
+  if ($prop->{sa_client}) {
+    $status = $self->{assassinc}->process(\$msglines);
+    return {
+      'isspam'    => $status->{isspam} eq "True",
+      'score'     => $status->{score},
+      'threshold' => $status->{threshold},
+      'message'   => $status->{message},
+      'report'    => $status->{report}
+    };
+  }
+  my $assassin = $self->{assassin};
+  my ($mail, $msg_resp);
+  if ($prop->{sa_version} >= 3) {
+    $mail = $assassin->parse(\$msglines, 0);
+  }
+  elsif ($prop->{sa_version} >= 2.70) {
+    $mail = Mail::SpamAssassin::MsgParser->parse(\$msglines);
+  }
+  else {
+    $mail = Mail::SpamAssassin::NoMailAudit->new(data => \$msglines);
+  }
+
+  # Check spamminess (returns Mail::SpamAssassin:PerMsgStatus object)
+  my $result = $assassin->check($mail);
+  # use Mail::SpamAssassin:PerMsgStatus object to rewrite message
+  if ($prop->{sa_version} >= 3) {
+    # inject _SPAMPDVERSION_ as a "template tag" (macro) for SA add_header
+    $result->set_tag("SPAMPDVERSION", $self->VERSION) if ($prop->{sa_version} >= 3.0020);
+    $msg_resp = $result->rewrite_mail;
+  }
+  else {
+    # SA versions prior to 3 need to get the response in a different manner
+    $result->rewrite_mail;
+    $msg_resp = join '', $mail->header, "\r\n", @{$mail->body};
+  }
+  $mail->finish();
+  return {
+    'isspam'    => $result->is_spam,
```

Comment:
Let's keep it as `is_spam` with the underscore separator.

Reply:
done